src/blenderbim/blenderbim/bim/module/aggregate/operator.py

```diff
@@ -201,18 +201,88 @@ class BIM_OT_select_aggregate(bpy.types.Operator):
     bl_label = "Select Aggregate"
     bl_options = {"REGISTER", "UNDO"}
     obj: bpy.props.StringProperty()
+    bl_description = (
+        "Selects the part's aggregate empties.\n\n"
+        + "SHIFT+CLICK - selects all the aggregate empties and all the parts within"
+    )
+
+    select_all: bpy.props.BoolProperty(name="Select All Parts", default=False, options={"SKIP_SAVE"})
+
+
+    def invoke(self, context, event):
+        if event.type == "LEFTMOUSE" and event.shift:
+            self.select_all = True
+        return self.execute(context)    
 
     def execute(self, context):
         self.file = IfcStore.get_file()
-        obj = bpy.data.objects.get(self.obj) or context.active_object
-        aggregate = ifcopenshell.util.element.get_aggregate(tool.Ifc.get_entity(obj))
-        aggregate_obj = tool.Ifc.get_object(aggregate)
-        if aggregate_obj in context.selectable_objects:
-            aggregate_obj.select_set(True)
-            bpy.context.view_layer.objects.active = aggregate_obj
+        #obj = bpy.data.objects.get(self.obj) or context.active_object
+        parts = []
+        for obj in context.selected_objects:
+            el = tool.Ifc.get_entity(obj)
+            if el:
+                aggregate = ifcopenshell.util.element.get_aggregate(el)
+                if aggregate:
+                    parts.append(aggregate)
+                    obj.select_set(False)
+                else:
+                    pass
+            if not el:
+               obj.select_set(False) 
+
+        #aggregate = ifcopenshell.util.element.get_aggregate(tool.Ifc.get_entity(obj))
+        #aggregate_obj = tool.Ifc.get_object(aggregate)
+        if self.select_all:
+            parts_objs = []
+            for part in parts:
+                if part.IsDecomposedBy:
+                    for subpart in part.IsDecomposedBy[0].RelatedObjects:
+                        parts.append(subpart)
+                parts_objs.append(part)
+
+            for element in parts_objs:
+                obj = tool.Ifc.get_object(element)
+                if obj:
+                    obj.select_set(True)
```

Comment:
There is no need to create a list, for the parts, you can select them in the end of loop. See the last comment example.